src/solvers/dg_simplices/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+- `du`, `u` are vectors
+- `ATr` is the transpose of the flux differencing matrix `A`. The transpose is used for 
+faster traversal since matrices are column major in Julia. 
+"""
+@inline @muladd function hadamard_sum_ATr!(du, ATr, volume_flux::VF, u, skip_index=(i,j)->false) where {VF}
+  rows,cols = axes(ATr)
+  for i in cols
+    ui = u[i]
+    val_i = du[i]
+    for j in rows
+      if !skip_index(i,j)                
+          val_i += ATr[j,i] * volume_flux(ui, u[j])
+      end
+    end
+    du[i] = val_i 
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{2, Tri}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  @unpack rxJ, sxJ, ryJ, syJ = mesh.md
+  QrskewTr, QsskewTr = Qrst_skew_Tr
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem] * QrskewTr + sxJ[1,elem] * QsskewTr))
+  else
+    return LazyArray(@~ @. 2 * (ryJ[1,elem] * QrskewTr + syJ[1,elem] * QsskewTr))
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{3, Tet}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  QrskewTr, QsskewTr, QtskewTr = Qrst_skew_Tr
+  @unpack rxJ, sxJ, txJ, ryJ, syJ, tyJ, rzJ, szJ, tzJ = mesh.md
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem]*QrskewTr + sxJ[1,elem]*QsskewTr + txJ[1,elem]*QtskewTr))
+  elseif orientation == 2
+    return LazyArray(@~ @. 2 * (ryJ[1,elem]*QrskewTr + syJ[1,elem]*QsskewTr + tyJ[1,elem]*QtskewTr))
+  elseif orientation == 3
+    return LazyArray(@~ @. 2 * (rzJ[1,elem]*QrskewTr + szJ[1,elem]*QsskewTr + tzJ[1,elem]*QtskewTr))
+  end
+end
+
+function calc_volume_integral!(du, u::StructArray, volume_integral::VolumeIntegralFluxDifferencing,
+                 mesh::VertexMappedMesh, equations, dg::DG, cache) where {DG <: SBPDGFluxDiff}
+
+  rd = dg.basis  
+  @unpack local_values_threaded = cache
+
+  volume_flux_oriented(i) = let i=i, equations=equations 
+    @inline (u_ll, u_rr)->volume_integral.volume_flux(u_ll, u_rr, i, equations)
+  end
+
+  # Todo: simplices. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+  @threaded for e in eachelement(mesh, dg, cache)
+    rhs_local = local_values_threaded[Threads.threadid()]
+    fill!(rhs_local, zero(eltype(rhs_local)))
+    u_local = view(u, :, e)
+    for i in eachdim(mesh) 
+      Qi_skew_Tr = build_lazy_physical_derivative(e, i, mesh, dg, cache)
+      hadamard_sum_ATr!(rhs_local, Qi_skew_Tr, volume_flux_oriented(i), u_local)
+    end
+    view(du, :, e) .+= rhs_local ./ rd.wq
+  end
+end
+
+
+function create_cache(mesh::VertexMappedMesh{Dim}, equations, dg::DG, 
+                      RealT, uEltype) where {DG <: PolyDGFluxDiff} where {Dim}
+
+  rd = dg.basis
+  @unpack md = mesh
+  
+  # Todo: simplices. Fix this when StartUpDG v0.11.0 releases: new API `Qrst_hybridized, VhP, Ph = StartUpDG.hybridized_SBP_operators(rd)`
```

Comment:
Can this be deleted?

Reply:
Do you mean the extra comment on StartUpDG v0.11.0? If so, sure.

Reply:
On second thought, it's better to keep this here. Any plans when to release v0.11 of StartUpDG.jl?

Reply:
Sure - hopefully by end of July.

Reply:
I think this should be kept. Once SUPDG is released, it will make the code dimension agnostic (I think).